mail_activity_done/models/res_users.py

```diff
@@ -38,10 +38,28 @@ def systray_get_activities(self):
             },
         )
         activity_data = self.env.cr.dictfetchall()
+        model_ids = [a["id"] for a in activity_data]
+        model_names = {
+            n[0]: n[1] for n in self.env["ir.model"].sudo().browse(model_ids).name_get()
+        }
         records_by_state_by_model = defaultdict(
             lambda: {"today": set(), "overdue": set(), "planned": set(), "all": set()}
         )
         for data in activity_data:
+            if not user_activities.get(data["res_model"]):
```

Comment:
It seems to me that if this is the case, none of the retrieved records that the activities are linked to are accessible to the current user (as per https://github.com/odoo/odoo/blob/f17428b/addons/mail/models/res_users.py#L178-L182), so you can just `continue` here.

Reply:
@StefanRijnhart Thanks, very good catch!
I reverted and pushed to proper fix now.
Please review and let me know.